### ISSUE TITLE
Weird inversion and res stuck at 240x240 on ESP32-S3-WROOM-1

### ISSUE THREAD

**Issue by 6b32** [Bodmer/TFT_eSPI#1899]
Hi, I upgraded from previous ESP32-WROOM-32U to this newer one however these issues emerged:
1. I need to use #define TFT_INVERSION_ON for tft.fillScreen(TFT_BLACK); to do black, otherwise is white. Which is fine. But text of color gets inverted or rather not inverted. So while background is correctly black now text instead of orange is blue.
2.  I am using 320x240 "wide" LCD, worked on previous setup, after much much trying. Now does not work again, usable area is capped at 240x240 no matter how these are: #define TFT_WIDTH  240  #define TFT_HEIGHT 320. Rest of the screen is gray noise. 
3. Also tft.setRotation(2); has chagned, not an issue but numbers now do different things.

Many Thanks for looking into this.

**Comment by 6b32** [comment]
Yes, SPI running same 40MHz as before.

**Comment by Bodmer** [comment]
As noted above, without a more detailed knowledge of your hardware and software it is difficult to advise further.

**Comment by mtojek** [comment]
Hi @Bodmer,

I'm facing the same problem, I guess that you can reopen the issue. TFT LCD bought at Shoppe, should be compatible with ILI9341.

Code used:

```c
#include <TFT_eSPI.h> // Graphics and font library for ILI9341 driver chip
#include <SPI.h>

TFT_eSPI tft = TFT_eSPI();  // Invoke library


void setup(void) {
  tft.init();
}

void loop() {
  // Fill screen with grey so we can see the effect of printing with and without 
  // a background colour defined
  tft.fillScreen(ILI9341_ORANGE);
  
  // Set "cursor" at top left corner of display (0,0) and select font 2
  // (cursor will move to next line automatically during printing with 'tft.println'
  //  or stay on the line is there is room for the text with tft.print)
  tft.setCursor(0, 0, 2);
  // Set the font colour to be white with a black background, set text size multiplier to 1
  tft.setTextColor(ILI9341_PURPLE);  tft.setTextSize(1);
  // We can now plot text on screen using the "print" class
  tft.println("It doesn't work.");

  delay(10000);
}
```

I'm using the following setup:

```c
#define USER_SETUP_ID 42

#define ILI9341_DRIVER
#define ILI9342_DRIVER
#define TFT_INVERSION_ON

#define TFT_MISO 19  // (leave TFT SDO disconnected if other SPI devices share MISO)
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS   15  // Chip select control pin
#define TFT_DC    2  // Data Command control pin
#define TFT_RST   4  // Reset pin (could connect to RST pin)

// Optional touch screen chip select
//#define TOUCH_CS 5 // Chip select pin (T_CS) of touch screen

#define LOAD_GLCD    // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2   // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4   // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6   // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7   // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8   // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF   // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

// TFT SPI clock frequency
// #define SPI_FREQUENCY  20000000
// #define SPI_FREQUENCY  27000000
 #define SPI_FREQUENCY  40000000
// #define SPI_FREQUENCY  80000000

// Optional reduced SPI frequency for reading TFT
#define SPI_READ_FREQUENCY  16000000

// SPI clock frequency for touch controller
#define SPI_TOUCH_FREQUENCY  2500000
```



![298974227_633673418188613_7967090700947346156_n](https://user-images.githubusercontent.com/14044910/185740826-94b08ee3-bfc4-4642-9689-0ed5d2200cb7.jpg)
![299325320_392273896381543_5620197420649163467_n](https://user-images.githubusercontent.com/14044910/185740831-ef32aaef-e130-4fda-babe-1b1c69b5f09e.jpg)

**Comment by mtojek** [comment]
Hey @6b32, did you solve your problem?

**Comment by mtojek** [comment]
Hey @6b32, did you solve your problem?

**Comment by 6b32** [comment]
> Hey @6b32, did you solve your problem?

Hey, sadly not so I moved away to another display which works. Also I see you are having troubles with getting full 320x240, "solved" that by some random resolution config in user setup and then set.rotation, in one coincidence it miraculously work in full screen mode.

**Comment by Bodmer** [comment]
Try setting the display resolution in the setup file.

#define TFT_WIDTH  320
#define TFT_HEIGHT 240

**Comment by 6b32** [comment]
Of course, tried that but it did nothing, still fixed in 240x240. It only worked when this was set to either 320x240 or 240x320 and only in one rotation except default (1) of course.

**Comment by Bodmer** [comment]
Sounds like the display has a ILI9342 driver chip, so use:
#define ILI9342_DRIVER

**Comment by mtojek** [comment]
I used setup 42 with minor modifications:

```
#define ILI9342_DRIVER
#define TFT_RGB_ORDER TFT_RGB
```

(I had to comment `#define ILI9341_DRIVER` everywhere else)

LCD shows colors correctly. Do you think that we should add another setup to select?

**Comment by BurianMartin** [comment]
Hello. 
I appear to be very late to this issue, but I ddi struggle just now with this and for a few hours tried to figure out what was wrong. 
In some time I figured out that in my case, it was a wrong driver setting. I was using the ILI9341driver, but when I switched to the HX8357D driver, my troubles were just gone. Display shows colors correctly as well as the 240x240 pixel problem is gone. Maybe some minor debugging in terms of screen mirroring and stuff, but that is a formality. 
Hope this helps someone.
